src/main/java/io/confluent/connect/hdfs/wal/FSWAL.java

```diff
@@ -60,7 +61,7 @@ public void append(String tempFile, String committedFile) throws ConnectExceptio
       writer.append(key, value);
       writer.hsync();
     } catch (IOException e) {
-      throw new ConnectException(e);
+      throw new DataException(e);
```

Comment:
Same questions about retriable errors here. I think there are cases where you could fail to acquire the WAL and given the way the state machine works, you'd want a retriable exception. Even if we weren't doing that correctly for all (or any) of these cases, `DataException` seems weird for at least some of the `IOExceptions` that can happen in the `FSWAL` code.